xds/internal/resolver/serviceconfig_test.go

```diff
@@ -43,3 +50,56 @@ func (s) TestPruneActiveClusters(t *testing.T) {
 		t.Fatalf("r.activeClusters = %v; want %v\nDiffs: %v", r.activeClusters, want, d)
 	}
 }
+
+// TestGenerateRequestHashHeaders tests generating request hashes for hash policies that specify
+// to hash headers.
+func (s) TestGenerateRequestHashHeaders(t *testing.T) {
+	cs := &configSelector{}
+
+	rpcInfo := iresolver.RPCInfo{
+		Context: metadata.NewIncomingContext(context.Background(), metadata.Pairs(":path", "/products")),
+		Method:  "/some-method",
+	}
+
+	hashPolicies := []*xdsclient.HashPolicy{
```

Comment:
Nit: delete the `\n`, so this line is `[]*xdsclient.HashPolicy{{`.

Reply:
Done.